### ISSUE TITLE
Is attention_mask supposed to be added to attention_weights? Based on functions docstring mask values are either 0 or 1.  

### ISSUE THREAD

**Issue by rpanackal** [huggingface/transformers#24897]
https://github.com/huggingface/transformers/blame/476be08c4aa96f8c1cae4200d2677bbe8f12cf80/src/transformers/models/autoformer/modeling_autoformer.py#L619

**Comment by sgugger** [comment]
cc @kashif

**Comment by kashif** [comment]
thanks @sgugger and @rpanackal let me check

**Comment by ArthurZucker** [comment]
Hey @rpanackal, the attention masks are first processed with:
```python
def _expand_mask(mask: torch.Tensor, dtype: torch.dtype, tgt_len: Optional[int] = None):
    """
    Expands attention_mask from `[bsz, seq_len]` to `[bsz, 1, tgt_seq_len, src_seq_len]`.
    """
    bsz, src_len = mask.size()
    tgt_len = tgt_len if tgt_len is not None else src_len

    expanded_mask = mask[:, None, None, :].expand(bsz, 1, tgt_len, src_len).to(dtype)

    inverted_mask = 1.0 - expanded_mask

    return inverted_mask.masked_fill(inverted_mask.to(torch.bool), torch.finfo(dtype).min)
```
so the value are no longer 0 and 1 😉
